### PR TITLE
chore(CPlusPlus): add palindrome

### PR DESCRIPTION
**Pull request by kumarsyadav2**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [x] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

adding a program to find palindrome number or string in c++
**Other information:**

### CODE REVIEW

**Comment by kumarsyadav2** [comment]
Why this didn't merged yet? If there's any problem then kindly let me know...
Did I have made pull request/issue in correct way or any it can be made into some other good ways...?

**Comment by ming-tsai** [comment]
> Why this didn't merged yet? If there's any problem then kindly let me know...
> Did I have made pull request/issue in correct way or any it can be made into some other good ways...?

We have to wait @UG-SEP to review this pull request

**Comment by ming-tsai** [comment]
Hi @kumarsyadav2,
If @UG-SEP does not response tomorrow I will assign another reviewer

**Comment by ming-tsai** [comment]
Hi @kumarsyadav2,
Please add a reference to these files:
- [en](https://github.com/kumarsyadav2/DSA/blob/main/docs/en/Strings/Palindrome.md)
- [zh-tw](https://github.com/kumarsyadav2/DSA/blob/main/docs/zh-tw/Strings/Palindrome.md)
- [ja](https://github.com/kumarsyadav2/DSA/blob/main/docs/ja/Strings/Palindrome.md)

**Comment by kumarsyadav2** [comment]
> Hi @kumarsyadav2,
> Please add a reference to these files:
> 
> * [en](https://github.com/kumarsyadav2/DSA/blob/main/docs/en/Strings/Palindrome.md)
> * [zh-tw](https://github.com/kumarsyadav2/DSA/blob/main/docs/zh-tw/Strings/Palindrome.md)
> * [ja](https://github.com/kumarsyadav2/DSA/blob/main/docs/ja/Strings/Palindrome.md)

Hi, i only know english language, i didn't understand file's language that is mentioned above!
also i am bit beginner in open source i.e. i don't know what reference to add in above files and how?
if there's any documentation or readme file that expalins what and how to add reference then kindly share with me... or guide me how to do. i will definitely add reference to these files.

**Comment by ming-tsai** [comment]
> > Hi @kumarsyadav2,
> > Please add a reference to these files:
> > 
> > * [en](https://github.com/kumarsyadav2/DSA/blob/main/docs/en/Strings/Palindrome.md)
> > * [zh-tw](https://github.com/kumarsyadav2/DSA/blob/main/docs/zh-tw/Strings/Palindrome.md)
> > * [ja](https://github.com/kumarsyadav2/DSA/blob/main/docs/ja/Strings/Palindrome.md)
> 
> Hi, i only know english language, i didn't understand file's language that is mentioned above!
> also i am bit beginner in open source i.e. i don't know what reference to add in above files and how?
> if there's any documentation or readme file that expalins what and how to add reference then kindly share with me... or guide me how to do. i will definitely add reference to these files.
> 

Ok, don't worries I will added another language by myself

**Comment by kumarsyadav2** [comment]
Let me try, whether i can do that or not.
initially it seems that i can't do this but after reading some **docs** and online stuff, it seems i can do that

**Comment by kumarsyadav2** [comment]
@ming-tsai kindly assign it active reviewer/member!!! They are not responding from last 3 days.

File: algorithms/CPlusPlus/Maths/palindrome.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#include<iostream>
+using namespace std;
+int main()
```
Comment (aayushjain7) [inline algorithms/CPlusPlus/Maths/palindrome.cpp:3]: provide description for program

File: algorithms/CPlusPlus/Maths/palindrome.cpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+	string new_str = "";
+
+	for(int x = (num_str.size()-1); x >= 0; x--){
+		new_str += num_str[x];
+	}
+
+	cout << (num_str == new_str ? "palindrome" : "Non-palindrome");
+
+	return 0;
+}
```
Comment (aayushjain7) [inline algorithms/CPlusPlus/Maths/palindrome.cpp:16]: provide test cases and time complexity

**Review by aayushjain7 — CHANGES_REQUESTED** [review]

**Review by kumarsyadav2 — COMMENTED** [review]
Hey @aayushjain7 , I added description, comments, time complexities and test-cases in code as comments.

**Review by aayushjain7 — APPROVED** [review]
nice!

**Pull request merged**